pkg/cgroups/cgroupsv2/fsutils.go

```diff
@@ -0,0 +1,153 @@
+package cgroupsv2
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+	"time"
+
+	jobv1 "github.com/kralicky/jobserver/pkg/apis/job/v1"
+	"golang.org/x/sys/unix"
+)
+
+func listControllers(file string) ([]string, error) {
+	info, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	info = bytes.TrimSpace(info)
+	return strings.Fields(string(info)), nil
+}
+
+func enableController(file, name string) (retErr error) {
+	f, err := os.OpenFile(file, os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintf(f, "+%s\n", name)
+	return errors.Join(err, f.Close())
+}
+
+const (
+	CFSPeriod   = 100000
+	CFSMinQuota = 1000
+)
+
+var availableMilliCpus = int64(runtime.NumCPU() * 1000)
```

Comment:
Absolutely - in a real system we would definitely want to check the environment of wherever the jobs are actually going to be run, since it affects the quota ratio calculation. I assume it's fine to hand-wave this? Or if you want, I can stub out some abstractions.